query-engine/query-engine-wasm/Cargo.toml

```diff
@@ -46,3 +46,15 @@ tracing-subscriber = { version = "0.3" }
 tracing-futures = "0.2"
 tracing-opentelemetry = "0.17.3"
 opentelemetry = { version = "0.17"}
+
+[package.metadata.wasm-pack.profile.release]
+wasm-opt = [
+    "-Os",                                # focus on code size
+    "--vacuum",                           # removes obviously unneeded code
+    "--duplicate-function-elimination",   # removes duplicate functions 
+    "--duplicate-import-elimination",     # removes duplicate imports
+    "--optimize-for-js",                  # early optimize of the instruction combinations for js
+    "--gsi",                              # globally optimize struct values
+    "--directize",                        # turns indirect calls into direct ones
+    "--cfp",                              # propagate constant struct field values
+]
```

Comment:
TODO: extract the "optimize" flags, and place them into a new draft PR for "sometimes later in the future". We agreed on focusing on size first, performance later.